src/Microsoft.Health.Dicom.Functions/DurableFunctionsCleanup.cs

```diff
@@ -0,0 +1,55 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using EnsureThat;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using System.Threading;
+
+namespace Microsoft.Health.Dicom.Functions
+{
+    public static class DurableFunctionsCleanup
+    {
+        // Currently an HTTP trigger, TODO: change to the appropriate trigger.
+        [FunctionName("DurableFunctionsCleanup")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req,
+            [DurableClient] IDurableOrchestrationClient client,
+            ILogger log)
+        {
+            log.LogInformation("Durable Functions cleanup started.");
+
+            EnsureArg.IsNotNull(req, nameof(req));
+
+            EnsureArg.IsNotNull(client, nameof(client));
+
+            // Specify conditions for Function Queries.
+            OrchestrationStatusQueryCondition condition = new OrchestrationStatusQueryCondition();
+
+            using (CancellationTokenSource tokenSource = new CancellationTokenSource())
+            {
+                EnsureArg.IsNotNull(tokenSource, nameof(tokenSource));
+                CancellationToken token = tokenSource.Token;
+
+                OrchestrationStatusQueryResult listOfOrchestrators = await client.ListInstancesAsync(condition, token);
+
+                // Loop through the orchestration instances and purge them.
+                foreach (var orch in listOfOrchestrators.DurableOrchestrationState)
+                {
+                    await client.PurgeInstanceHistoryAsync(orch.InstanceId);
+                }
```

Comment:
As this:
```
        [Obsolete]
        Task<IList<DurableOrchestrationStatus>> GetStatusAsync(DateTime? createdTimeFrom = null, DateTime? createdTimeTo = null, IEnumerable<OrchestrationRuntimeStatus> runtimeStatus = null, CancellationToken cancellationToken = default);
```
is obsolete, we use this:
```
        Task<OrchestrationStatusQueryResult> ListInstancesAsync(OrchestrationStatusQueryCondition condition, CancellationToken cancellationToken);
```

However, the current purge instance history has similar parameters as the obsolete function to get orchestration instances.
```
Task<PurgeHistoryResult> PurgeInstanceHistoryAsync(DateTime createdTimeFrom, DateTime? createdTimeTo, IEnumerable<OrchestrationStatus> runtimeStatus);
```
This is why I loop through the orchestrators.